### PR TITLE
Rename ApiSerializer's mutate to attributes

### PR DESCRIPTION
**Pull request by SychO9**
**Part of #1891**

**Confirmed**

- [x] Backend changes: tests are green (run `composer test`).

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)

### CODE REVIEW

File: src/Extend/ApiSerializer.php [hunk 1]
```diff
@@ -67,13 +67,35 @@ public function attribute(string $name, $callback)
      *
      * @return self
      */
-    public function mutate($callback)
+    public function attributes($callback)
     {
         $this->mutators[] = $callback;
```
Comment (askvortsov1) [inline src/Extend/ApiSerializer.php:72]: Lets rename this property too

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

File: src/Extend/ApiSerializer.php [hunk 2]
```diff
@@ -145,10 +167,10 @@ public function extend(Container $container, Extension $extension = null)
             };
         }
 
-        foreach ($this->mutators as $mutator) {
-            $mutator = ContainerUtil::wrapCallback($mutator, $container);
+        foreach ($this->attributes as $callback) {
+            $callback = ContainerUtil::wrapCallback($callback, $container);
 
-            AbstractSerializer::addMutator($this->serializerClass, $mutator);
+            AbstractSerializer::addMutator($this->serializerClass, $callback);
```
Comment (SychO9) [inline src/Extend/ApiSerializer.php:173]: do we want to rename the method as well as the property in `AbstractSerializer` while we're at it as well ?

Comment (askvortsov1) [reply src/Extend/ApiSerializer.php:173]: Yep! Especially since that's not really public API.

File: src/Api/Serializer/AbstractSerializer.php [hunk 3]
```diff
@@ -43,7 +43,7 @@ abstract class AbstractSerializer extends BaseAbstractSerializer
     /**
      * @var callable[]
      */
-    protected static $mutators = [];
+    protected static $customAttributes = [];
```
Comment (askvortsov1) [inline src/Api/Serializer/AbstractSerializer.php:46]: `customAttributes` doesn't feel plural enough IMO. Maybe something like `attributeMutator`?

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

File: src/Api/Serializer/AbstractSerializer.php (hunk 3 above)
Comment (SychO9) [reply src/Api/Serializer/AbstractSerializer.php:46]: wouldn't it be `attributeMutators` then? since it's many

Comment (askvortsov1) [reply src/Api/Serializer/AbstractSerializer.php:46]: Yeah that sounds good!

**Review by askvortsov1 — APPROVED** [review]

**Review by tankerkiller125 — APPROVED** [review]

File: src/Extend/ApiSerializer.php [hunk 4]
```diff
@@ -67,13 +67,35 @@ public function attribute(string $name, $callback)
      *
      * @return self
      */
-    public function mutate($callback)
+    public function attributes($callback)
```
Comment (askvortsov1) [inline src/Extend/ApiSerializer.php:70]: I just realized, we need to deprecate the `mutate` method, since that was released in the previous version

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

File: src/Extend/ApiSerializer.php (hunk 4 above)
Comment (SychO9) [reply src/Extend/ApiSerializer.php:70]: it is deprecated, look below this code

Comment (askvortsov1) [reply src/Extend/ApiSerializer.php:70]: I.... don't know how I missed that... What.... :see_no_evil:

**Review by askvortsov1 — APPROVED** [review]

**Pull request merged**